moped-editor/src/views/projects/newProjectView/DefineProjectForm.js

```diff
@@ -1,102 +1,198 @@
-  descriptionError,
-  setSignalRecord,
-  fromSignalAsset,
-  setFromSignalAsset,
-  signal,
-  setSignal,
-  signalError,
-}) => {
-  const handleFieldChange = (value, name) => {
-    const updatedProjectDetails = { ...projectDetails, [name]: value };
+import {
+  getSignalOptionLabel,
+  getSignalOptionSelected,
+  SOCRATA_ENDPOINT,
+} from "src/utils/signalComponentHelpers";
 
-    setProjectDetails(updatedProjectDetails);
-  };
+const validationSchema = yup.object().shape({
+  projectName: yup.string().nullable().optional().when("isSignalProject", {
+    is: false,
+    then: agolValidation.projectName,
+  }),
+  projectSecondaryName: agolValidation.projectSecondaryName,
+  description: agolValidation.description,
+  signal: yup
+    .object()
+    .nullable()
+    .optional()
+    .when("isSignalProject", {
+      is: true,
+      then: yup.object().required("Required"),
+    }),
+  isSignalProject: yup.boolean(),
+});
+
+/**
+ * DefineProjectForm Component
+ * @param {Function} handleSave - The function to save the new project
+ * @param {Boolean} loading - Loading state to show form submission
+ * @param {Boolean} success - Success state to show successful form submission
+ * @returns {JSX.Element}
+ * @constructor
+ */
+const DefineProjectForm = ({ handleSave, loading, success }) => {
+  const {
+    handleSubmit,
+    control,
+    watch,
+    formState: { errors, isDirty, isValid },
+  } = useForm({
+    defaultValues: {
+      projectName: null,
+      projectSecondaryName: null,
+      description: null,
+      isSignalProject: false,
+      signal: null,
+    },
+    mode: "onChange",
+    resolver: yupResolver(validationSchema),
+  });
+
+  const [isSignalProject] = watch(["isSignalProject"]);
 
   return (
-    <form style={{ padding: 25 }}>
-      <Grid container spacing={3} style={{ margin: 20 }}>
-        <Grid item xs={6}>
-          {!fromSignalAsset && ( // if
-            <TextField
-              sx={{ marginBottom: "2rem" }}
-              variant="standard"
-              required
-              autoFocus
-              label="Project name"
-              name="project_name"
-              type="text"
-              fullWidth
-              value={projectDetails.project_name}
-              error={nameError}
-              helperText="Required"
-              InputLabelProps={{ required: false }}
-              onChange={(e) => handleFieldChange(e.target.value, e.target.name)}
-            />
-          )}
-          {fromSignalAsset && ( // else
-            <Box sx={{ marginBottom: "2.1rem" }}>
-              <SignalAutocomplete
-                signal={signal}
-                setSignal={setSignal}
-                projectDetails={projectDetails}
-                setProjectDetails={setProjectDetails}
-                setSignalRecord={setSignalRecord}
-                signalError={signalError}
+    <Box
+      component="form"
+      onSubmit={handleSubmit(handleSave)}
+      sx={{ paddingX: 9, paddingY: 9 }}
```

Comment:
is theme.spacing an option? 😬

Reply:
i originally had theme.spacing here, but then I remembered your TIL about the shorthand for theme spacing that is available when using the sx prop. [Link](https://mui.com/system/getting-started/the-sx-prop/?srsltid=AfmBOorFiXCdoDsefRZs_4LnCwuP3P0TVweUDxEJTozDLrCnmMUwGtgP#spacing). I have a feeling we will be TIL this over and over. 😵‍💫